src/main/java/com/hoverla/bring/context/CacheProxyConfigurator.java

```diff
@@ -0,0 +1,53 @@
+package com.hoverla.bring.context;
+
+import com.hoverla.bring.annotation.Cacheable;
+import net.sf.cglib.proxy.Enhancer;
+import net.sf.cglib.proxy.MethodInterceptor;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CacheProxyConfigurator implements ProxyConfigurator {
+
+    private Map<String, Map<List<Object>, Object>> cache = new ConcurrentHashMap<>();
+
+    @Override
+    public Object replaceWithProxyIfNeeded(Object t, Class<?> implClass) {
+        boolean isCacheable = Arrays.stream(implClass.getMethods()).anyMatch(m -> m.isAnnotationPresent(Cacheable.class));
+        if (isCacheable) {
+            MethodInterceptor methodInterceptor = (obj, method, args, proxy) -> {
+                if (method.isAnnotationPresent(Cacheable.class)) {
```

Comment:
Is it "if" redundant?

Reply:
no, first time it check if class has any method with 'Cacheble' annotation, and inside of method interceptor we check if method is annotated with cacheable we are going return value from cache another way we call method from original object